### ISSUE TITLE
Add an MLMHead layer

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#129]
This layer can go with the MLMMaskGenerator layer to make it easy to write a masked language model transformer.

The layer should map two inputs:
 - embedded tokens of shape `(batch, sequence_length, model_dim)`
 - mask positions of shape `(batch, prediction_length)`

To one output:
 - predictions of shape `(batch, prediction_length, vocabulary_size)`
 
 Where the outputs contains either logits or softmax probabilities per word in the vocabulary.

The layer should optionally allow reusing the embedding weights to reverse word embeddings from model_dim -> vocabulary_size.

**Comment by mattdangerw** [comment]
Rough flow of this layer:
 - `tf.gather` the masked word embeddings using the input positions
 -  add a dense layer from `model_dim` -> `model_dim`
 -  add another dense layers from `model_dim` -> `vocabulary_size`

This last dense layer is where we need to allow re-using the transposed embedding weights.

**Comment by aflah02** [comment]
Hey
I can take this up!

**Comment by mattdangerw** [comment]
@aflah02 thanks, but this one I already have :), made it while working on a guide for keras-nlp. We will open up more issues this week!

**Comment by aflah02** [comment]
Ohh!! right I just noticed that it's assigned to you. Sorry about the confusion!

**Comment by mattdangerw** [comment]
We will likely have a follow up issue here for RaggedTensor support, which could be a good one to learn about ragged tensors, but might also hit some rough edges in the tensorflow library. I can tag you for that and see if it's something you want to take on.

**Comment by aflah02** [comment]
Sure I'd be interested to check that out!
